### ISSUE TITLE
staticcheck 2022.1 does not work with golang 1.18

### ISSUE THREAD

**Issue by m15o** [Homebrew/homebrew-core#99654]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
brew config
HOMEBREW_VERSION: 3.4.7
ORIGIN: https://github.com/Homebrew/brew
HEAD: 09e766f83c8a420cce9e2210c46a24ec3864a094
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: aa599c38241323d2720fe4949cad0fe41c5d1c72
Core tap last commit: 35 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 13.1.6 build 1316
Git: 2.36.0 => /usr/local/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.3.1-x86_64
CLT: 13.3.1.0.1.1648687083
Xcode: 13.3.1

brew doctor
Your system is ready to brew.
```

### Verification

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

Trying to run staticcheck with go 1.18.1.
Because staticcheck 2022.1 includes go 1.18 support

### What happened (include all command output)?

Staticcheck complains about golang 1.18 feature.

```bash
cat > main.go <<EOF
package main

import "container/list"

func main() {
        list.New()
}
EOF

staticcheck main.go
```

output:
```
/usr/local/bin/staticcheck main.go
/usr/local/Cellar/go/1.18.1/libexec/src/container/list/list.go:27:8: undeclared name: any (compile)
/usr/local/Cellar/go/1.18.1/libexec/src/container/list/list.go:103:30: undeclared name: any (compile)
/usr/local/Cellar/go/1.18.1/libexec/src/container/list/list.go:134:35: undeclared name: any (compile)
/usr/local/Cellar/go/1.18.1/libexec/src/container/list/list.go:144:28: undeclared name: any (compile)
/usr/local/Cellar/go/1.18.1/libexec/src/container/list/list.go:150:27: undeclared name: any (compile)
/usr/local/Cellar/go/1.18.1/libexec/src/container/list/list.go:158:31: undeclared name: any (compile)
/usr/local/Cellar/go/1.18.1/libexec/src/container/list/list.go:169:30: undeclared name: any (compile)
```

### What did you expect to happen?

homebrew staticcheck works with soucecodes that contains golang 1.18 feature as official binary does
https://github.com/dominikh/go-tools/releases/tag/2022.1

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
1. Install golang and staticcheck

brew install go staticcheck

2. Create .go file

cat > main.go <<EOF
package main

import "container/list"

func main() {
        list.New()
}
EOF

3. Run staticcheck against the .go file.

staticcheck main.go
```

**Comment by dominikh** [comment]
What's the output of `/usr/local/bin/staticcheck -debug.version`, `go version` and `which go`?

**Comment by m15o** [comment]
The outputs of the commands are follows:

`/usr/local/bin/staticcheck -debug.version`
```
staticcheck 2022.1 (v0.3.0)

Compiled with Go version: go1.17.8
Main module:
        honnef.co/go/tools
Dependencies:
        github.com/BurntSushi/toml@v0.4.1 (sum: h1:GaI7EiDXDRfa8VshkTj7Fym7ha+y8/XxIgD2okUIjLw=)
        golang.org/x/exp/typeparams@v0.0.0-20220218215828-6cf2b201936e (sum: h1:qyrTQ++p1afMkO4DPEeLGq/3oTsdlvdH4vqZUBWzUKM=)
        golang.org/x/mod@v0.6.0-dev.0.20220106191415-9b9b3d81d5e3 (sum: h1:kQgndtyPBW/JIYERgdxfwMYh3AVStj88WQTlNDi2a+o=)
        golang.org/x/sys@v0.0.0-20211019181941-9d821ace8654 (sum: h1:id054HUawV2/6IGm2IV8KZQjqtwAOo2CYlOToYqa0d0=)
        golang.org/x/tools@v0.1.11-0.20220316014157-77aa08bb151a (sum: h1:ofrrl6c6NG5/IOSx/R1cyiQxxjqlur0h/TvbUhkH0II=)
        golang.org/x/xerrors@v0.0.0-20200804184101-5ec99f83aff1 (sum: h1:go1bK/D/BFZV2I8cIQd1NKEZ+0owSTG1fDTci4IqFcE=)
```
---
`go version`
```
go version go1.18.1 darwin/amd64
```
---
`which go`
```
/usr/local/bin/go
```

**Comment by dominikh** [comment]
The issue being this: `Compiled with Go version: go1.17.8` – it looks like Staticcheck wasn't built with Go 1.18. For it to support Go 1.18, it has to be built with Go 1.18 (or newer.)

**Comment by cho-m** [comment]
Should be fixed with #99695
